src/iconv.jl

```diff
@@ -188,10 +192,18 @@ function StringDecoder(istream::IO, from::ASCIIString, to::ASCIIString="UTF-8")
     s = StringDecoder(istream, cd, inbuf, outbuf,
                       Ref{Ptr{UInt8}}(pointer(inbuf)), Ref{Ptr{UInt8}}(pointer(outbuf)),
                       Ref{Csize_t}(0), Ref{Csize_t}(BUFSIZE), 0)
-    finalizer(s, close)
+    finalizer(s, finalize!)
```

Comment:
OK, I've been using a convention of adding the !, because it does modify the thing it finalizes.